File: wear/src/main/java/com/thewizrd/simplewear/controls/MusicPlayerViewModel.java
package com.thewizrd.simplewear.controls;

import android.graphics.Bitmap;

import java.util.Locale;

public class MusicPlayerViewModel {
    private Bitmap mBitmapIcon;
    private String mAppLabel;
    private String mPackageName;
    private String mActivityName;

    public String getKey() {
        if (mPackageName != null && mActivityName != null) {
            return String.format(Locale.ROOT, "%s/%s", mPackageName, mActivityName);
        }

        return null;
    }

    public Bitmap getBitmapIcon() {
        return mBitmapIcon;
    }

    public void setBitmapIcon(Bitmap bitmapIcon) {
        this.mBitmapIcon = bitmapIcon;
    }

    public String getAppLabel() {
        return mAppLabel;
    }

    public void setAppLabel(String label) {
        this.mAppLabel = label;
    }

    public String getPackageName() {
        return mPackageName;
    }

    public void setPackageName(String packageName) {
        this.mPackageName = packageName;
    }

    public String getActivityName() {
        return mActivityName;
    }

    public void setActivityName(String activityName) {
        this.mActivityName = activityName;
    }
}

File: wear/src/main/java/com/thewizrd/simplewear/controls/MusicPlayerViewModel.kt
package com.thewizrd.simplewear.controls

import android.graphics.Bitmap
import java.util.*

class MusicPlayerViewModel {
    var bitmapIcon: Bitmap? = null
    var appLabel: String? = null
    var packageName: String? = null
    var activityName: String? = null
    val key: String?
        get() {
            if (packageName != null && activityName != null) {
                return String.format(Locale.ROOT, "%s/%s", packageName, activityName)
            } else {
                return null
            }
        }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as MusicPlayerViewModel

        if (bitmapIcon != other.bitmapIcon) return false
        if (appLabel != other.appLabel) return false
        if (packageName != other.packageName) return false
        if (activityName != other.activityName) return false

        return true
    }

    override fun hashCode(): Int {
        var result = bitmapIcon?.hashCode() ?: 0
        result = 31 * result + (appLabel?.hashCode() ?: 0)
        result = 31 * result + (packageName?.hashCode() ?: 0)
        result = 31 * result + (activityName?.hashCode() ?: 0)
        return result
    }
}